emission/net/ext_service/transit_matching/match_stops.py

```diff
@@ -96,17 +97,28 @@ def get_predicted_transit_mode(start_stops, end_stops):
         return [rim.tags.route for rim in rel_id_matches]
 
     # Did not find matching routes. Let's see if stops are both "railway",
-    # if so, we can mark as TRAIN 
+    # if so, we can mark as TRAIN
     # TODO: return more complex kinds of railways?
-    p_start_train = ["railway" in s.tags for s in start_stops]
-    p_end_train = ["railway" in s.tags for s in end_stops]
-
-    logging.debug("len(start_train) = %s, len(end_train) = %s" % (
-        (len(p_start_train), len(p_end_train))))
-    if is_true(p_start_train) and is_true(p_end_train):
-        logging.debug("start and end are both TRAIN, returning TRAIN")
-        return ["TRAIN"]
-
+    p_start_train = [extract_railway_modes(s.tags) for s in start_stops]
+    p_start_train = Counter(itertools.chain.from_iterable(set(i) for i in p_start_train))
+    p_end_train = [extract_railway_modes(s.tags) for s in end_stops]
+    p_end_train = Counter(itertools.chain.from_iterable(set(i) for i in p_end_train))
+    logging.debug("len(start_train) = %d, len(end_train) = %d" %
+        (len(p_start_train), len(p_end_train)))
+    if len(p_start_train) > 0 and len(p_end_train) > 0:
+        p_start_max = p_start_train.most_common(1)[0][0]
+        p_end_max = p_end_train.most_common(1)[0][0]
+        if p_start_max is not None and p_start_max == p_end_max:
```

Comment:
Are you sure that building a check from the `most_common` is the best strategy? What if there is an overlapping mode, but it is not the most common? Would a set intersection work better? I know that @PatGendre said that he has taken tram trips, have you tried running the algorithm for any of them to see how well it works?

Reply:
When I tested on the data that I had (some trips in Berlin), I found that most of the time the `most_common` was the right mode but after tested it recently on new data in Marseille, it was not the case so I will try to use a set intersection.